pallets/polkapool/src/lib.rs

```diff
@@ -262,9 +262,14 @@ decl_module! {
             let seed = <T as Config>::RandomnessSource::random_seed();
             let mut rng = ChaChaRng::from_seed(*seed.0.as_fixed_bytes());
             stored_exts.store.shuffle(&mut rng);
+            total_weight = total_weight.saturating_add(base_weight);
             // Start executing
             for ext in stored_exts.store{
-                total_weight += ext.call.get_dispatch_info().weight;
+                total_weight = total_weight.saturating_add(ext.call.get_dispatch_info().weight);
+                if total_weight <= T::MaxAllowedWeight::get() {
```

Comment:
Makes sense

Reply:
> @Gauthamastro I think it should have breaked here if the `total_weight` is greater than `MaxAllowedWeight`?

Shouldn't it break after substrating the weight of the ext that we were supposed to execute but didn't execute since it will go over the allocated weight, since we didn't execute it, shouldn't we reduce that extra ext weight before the break.

Reply:
> > @Gauthamastro I think it should have breaked here if the `total_weight` is greater than `MaxAllowedWeight`?
> 
> Shouldn't it break after substrating the weight of the ext that we were supposed to execute but didn't execute since it will go over the allocated weight, since we didn't execute it, shouldn't we reduce that extra ext weight before the break.

Yes. The `break` is at the right place. I meant changing the if condition here which changes the breaking out of the loop logic.

Reply:
Ahh. alright